runtime/doc/quickfix.txt

```diff
@@ -138,11 +138,13 @@ processing a quickfix or location list command, it will be aborted.
 			current window is used instead of the quickfix list.
 
 							*:cq* *:cquit*
-:cq[uit][!]		Quit Vim with an error code, so that the compiler
-			will not compile the same file again.
-			WARNING: All changes in files are lost!  Also when the
-			[!] is not used.  It works like ":qall!" |:qall|,
-			except that Vim returns a non-zero exit code.
+:[count]cq[uit]		Quit Vim with an error code (When vim is called from 
+			another program, the caller might behave in a special way if the
+			error code is not 0; e.g. `git commit`, `fc` in bash, some
+			compilers.)
```

Comment:
Missing documentation on what “count” means, also this does not have to have parenthesis:

> Quit Vim with an error code, or the code specified in `[count]`. This may make a difference when Vim is called from another program which needs editor: e.g. `git commit` will abort the comitting process, `fc` (built-in from various shells including bash and zsh) will not execute the command, …

No need to mention “some compilers”, it is not useful without some specifics (like what was in original message). Not suggesting to add “compiler will not compile …” because of me never hearing the compiler which bothers to call editor at all.